runtime/runc_jailer.go

```diff
@@ -138,10 +138,18 @@ func newRuncJailer(
 
 func (j *runcJailer) prepareBindMounts(mounts []*proto.FirecrackerDriveMount) error {
 	for _, m := range mounts {
-		err := j.bindMountFileToJail(m.HostPath, filepath.Join(j.RootPath(), m.HostPath))
-		if err != nil {
+		stat := syscall.Stat_t{}
+		if err := syscall.Stat(m.HostPath, &stat); err != nil {
 			return err
 		}
+		// Only bindMount regular file.
+		// To avoid duplicate files, for block device, we will use system call to create device file for it.
+		if stat.Mode&syscall.S_IFMT == syscall.S_IFREG {
```

Comment:
I'm a bit confused about what we are doing here.

- For regular files, calling bindMoutFileToJail() is fine since ExposeFileToJail() is ultimately doing the same
- For block devices, calling bindMoutFileToJail() is problematic since it would break ExposeFileToJail().

Is it correct understanding? If we ultimately call ExposeFileToJail(), what is the responsibility of prepareBindMounts()? Sorry I might be the one who wrote this function though.

Also, please write some tests around your change.

Reply:
I will add more details on this.

Reply:
I was actually thinking to remove prepareBindMount() before. I was not quite confident to remove it given the below commit. For better or worse, I decided to work around. If prepareBindMount() could be possibly used for a feature developing in the future, we'd better to keep it, I just don't have the broad picture where it should go.
https://github.com/firecracker-microvm/firecracker-containerd/commit/666cb5e5a3ecc2a6bf286c56ab791c0a2d3d6201